### DISCUSSION TITLE
Displaying database and schema names when selecting dataset

### DISCUSSION THREAD

**Discussion by soniagtm** [apache/superset#25545] [Ideas]
Hello everyone, 

I would like to discuss about the feature I have implemented and wish to contribute to the community.

**Background**

When we want to create a new chart, we have to choose a dataset from the dropdown options. The problem is that when we have multiple datasets with identical names, it is hard to identify the correct dataset that we want to use to create chart.
 
The same issue is encountered when we attempt to select a dataset for adding a filter to the dashboard.
 
Here are some screenshots to illustrate the problem I am describing:

_**On the chart creation page**_

<img width="500" alt="Screenshot 2566-10-05 at 16 39 15" src="https://github.com/apache/superset/assets/96282005/86ef19d5-eb44-49d7-a251-2f98f2f0edd9"><br><br>

_**Add and edit filters to dashboard**_<br>
<img width="500" alt="Screenshot 2566-10-05 at 16 42 07" src="https://github.com/apache/superset/assets/96282005/23e35263-15a7-4f2a-900e-5ae2a04f65c4">

From the pictures, we can see that there are 4 datasets with the same name, 'Users'. Accordingly, if we want to choose the 'Users' dataset created from the Impala database, we would not be able to differentiate it unless we modify the dataset name.

---

**Solution**
 
The solution I have implemented is to add database and schema names beneath the dataset name, along with a tooltip label, so that users can identify the database and schema from which each dataset is created.

For datasets with a null schema name, the tooltip label will display ‘Not defined’ as the schema name.
 
Here are some screenshots displaying the solution I have implemented:

_**On the chart creation page**_

<img width="500" alt="Screenshot 2566-10-05 at 22 58 50" src="https://github.com/apache/superset/assets/96282005/5bde5c78-862e-40af-baf0-30a46d96ab5c"><br><br>

_**Add and edit filters to dashboard**_<br>
<img width="500" alt="Screenshot 2566-10-05 at 23 00 01" src="https://github.com/apache/superset/assets/96282005/332c03c1-ce2b-4339-b9d6-ca10e24ebb80">

 
I would appreciate any feedback or suggestions to this enhancement.
Thanks!
<br>

**Comment by felixuko** [comment]
@soniagtm was this finally done?
